### PR TITLE
Fix Outbound Provisioning Connector Configure issue

### PR DESCRIPTION
**Pull request by JayaShakthi97**
### Purpose
- Fix "Something went wrong" issue when configuring outbound provisioning connector (Initial purpose)
- Fix "Something went wrong" issue when listing configured outbound provisioning connector
- Fix not functioning accordion UI expansion
- Refactor to fix eslint warnings 

### Related Issues
- https://github.com/wso2/product-is/issues/12973

### Checklist
- [ ] e2e cypress tests locally verified.
- [x] Manual test round performed and verified.
- [ ] UX/UI review done on final implementation.
- [ ] Documentation provided. (Add links if there's any)
- [ ] Unit tests provided. (Add links if there's any)
- [ ] Integration tests provided. (Add links if there's any)

### Related PRs
- None

### Security checks
- [ ] Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines?
- [ ] Ran FindSecurityBugs plugin and verified report?
- [ ] Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets?

### CODE REVIEW

File: apps/console/src/features/identity-providers/components/settings/outbound-provisioning-settings.tsx [hunk 1]
```diff
@@ -321,6 +325,30 @@ export const OutboundProvisioningSettings: FunctionComponent<ProvisioningSetting
...
+     *
+     * @param {React.SyntheticEvent} e - Click event.
+     * @param {AccordionTitleProps} SegmentedAuthenticatedAccordion - Clicked title.
+     */
+    const handleAccordionOnClick = (e: MouseEvent<HTMLDivElement>,
+        SegmentedAuthenticatedAccordion: AccordionTitleProps): void => {
+        if (!SegmentedAuthenticatedAccordion) {
+            return;
+        }
+        const newIndexes = [ ...accordionActiveIndexes ];
```
Comment (brionmario) [inline apps/console/src/features/identity-providers/components/settings/outbound-provisioning-settings.tsx:339]:
```suggestion

        const newIndexes = [ ...accordionActiveIndexes ];
```

File: apps/console/src/features/identity-providers/components/settings/outbound-provisioning-settings.tsx [hunk 2]
```diff
@@ -321,6 +325,30 @@ export const OutboundProvisioningSettings: FunctionComponent<ProvisioningSetting
...
+
+        if (newIndexes.includes(SegmentedAuthenticatedAccordion.accordionIndex)) {
+            const removingIndex = newIndexes.indexOf(SegmentedAuthenticatedAccordion.accordionIndex);
```
Comment (brionmario) [inline apps/console/src/features/identity-providers/components/settings/outbound-provisioning-settings.tsx:342]:
```suggestion
            const removingIndex: number = newIndexes.indexOf(SegmentedAuthenticatedAccordion.accordionIndex);
```

File: apps/console/src/features/identity-providers/components/settings/outbound-provisioning-settings.tsx [hunk 3]
```diff
@@ -321,6 +325,29 @@ export const OutboundProvisioningSettings: FunctionComponent<ProvisioningSetting
...
+     * Handles accordion title click.
+     *
+     * @param {React.SyntheticEvent} e - Click event.
+     * @param {AccordionTitleProps} data - Clicked title.
+     */
+    const handleAccordionOnClick = (e: MouseEvent<HTMLDivElement>, data: AccordionTitleProps): void => {
+        if (!data) {
+            return;
+        }
+        const newIndexes: number[] = [ ...accordionActiveIndexes ];
```
Comment (brionmario) [inline apps/console/src/features/identity-providers/components/settings/outbound-provisioning-settings.tsx:338]:
```suggestion
        
        const newIndexes: number[] = [ ...accordionActiveIndexes ];
```

**Review by brionmario — APPROVED** [review]

File: apps/console/src/features/identity-providers/components/settings/outbound-provisioning-settings.tsx [hunk 4]
```diff
@@ -321,6 +325,30 @@ export const OutboundProvisioningSettings: FunctionComponent<ProvisioningSetting
...
     };
 
+    /**
+     * Handles accordion title click.
+     *
+     * @param {React.SyntheticEvent} e - Click event.
+     * @param {AccordionTitleProps} SegmentedAuthenticatedAccordion - Clicked title.
+     */
+    const handleAccordionOnClick = (e: MouseEvent<HTMLDivElement>,
+        SegmentedAuthenticatedAccordion: AccordionTitleProps): void => {
```
[the remark this answers is not in the document]
Comment (JayaShakthi97) [reply apps/console/src/features/identity-providers/components/settings/outbound-provisioning-settings.tsx]: Fixed with 0f97957893e6678fa5511341d7e8c106c474dbd0

File: apps/console/src/features/identity-providers/components/settings/outbound-provisioning-settings.tsx (hunk 1 above)
Comment (JayaShakthi97) [reply apps/console/src/features/identity-providers/components/settings/outbound-provisioning-settings.tsx]: Fixed with 0f97957893e6678fa5511341d7e8c106c474dbd0

File: apps/console/src/features/identity-providers/components/settings/outbound-provisioning-settings.tsx (hunk 2 above)
Comment (JayaShakthi97) [reply apps/console/src/features/identity-providers/components/settings/outbound-provisioning-settings.tsx]: Fixed with 0f97957893e6678fa5511341d7e8c106c474dbd0

File: apps/console/src/features/identity-providers/components/settings/outbound-provisioning-settings.tsx (hunk 3 above)
Comment (JayaShakthi97) [reply apps/console/src/features/identity-providers/components/settings/outbound-provisioning-settings.tsx:339]: Fixed with 848627d0115b990f947e8c0806fea5939220a5a6

File: apps/console/src/features/identity-providers/components/settings/outbound-provisioning-settings.tsx [hunk 5]
```diff
@@ -331,11 +359,11 @@ export const OutboundProvisioningSettings: FunctionComponent<ProvisioningSetting
...
-                (!isLoading)
-                    ? (
-                        <div className="default-provisioning-connector-section">
-                            <Grid>
-                                <Grid.Row>
+                    (!isLoading)
+                        ? (
+                            <div className="default-provisioning-connector-section">
+                                <Grid>
+                                    <Grid.Row>
```
Comment (Yasara-Y) [inline apps/console/src/features/identity-providers/components/settings/outbound-provisioning-settings.tsx:366]: Is the indentation correct for <Grid.Row> and <Grid> ?

**Pull request merged**